### PR TITLE
Implemented UnixChangeFileMode for windows.

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot Be a nice bot once again, test please!

**Comment by mfleisz** [comment]
+1

**Comment by bmiklautz** [comment]
@akallabeth nice. +1 from my side too. Little suggestion as far as I know _wchmod only supports read and write flags. Maybe we should fail in all other cases?

**Comment by akallabeth** [comment]
@bmiklautz Ok, implemented a warning log message for unsupported file mode constants.

File: winpr/libwinpr/file/generic.c [hunk 1]
```diff
@@ -828,6 +830,21 @@ int UnixChangeFileMode(const char* filename, int flags)
...
-	return 0;
+	int rc;
+	WCHAR *wfl = NULL;
+
+	int fl = 0;
+
+	ConvertToUnicode(CP_UTF8, 0, filename, -1, &wfl, 0);
+	if (!wfl)
+		return -1;
+
```
Comment (bmiklautz) [inline winpr/libwinpr/file/generic.c]: Although this check works it was suggested to check the return value of ConvertToUnicode.

**Comment by bmiklautz** [comment]
+1

**Comment by bmiklautz** [comment]
@freerdp-bot re-test!

**Pull request merged**